core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -310,53 +320,44 @@ public int onDataRead(
         final long transferredLength = decodedReq.transferredBytes();
         if (maxContentLength > 0 && transferredLength > maxContentLength) {
             assert encoder != null;
-            final Http2Stream stream = encoder.findStream(streamId);
-            if (isWritable(stream)) {
-                final ContentTooLargeException cause =
-                        ContentTooLargeException.builder()
-                                                .maxContentLength(maxContentLength)
-                                                .contentLength(req.headers())
-                                                .transferred(transferredLength)
-                                                .build();
-
-                writeErrorResponse(streamId, req.headers(), HttpStatus.REQUEST_ENTITY_TOO_LARGE, null, cause);
-                decodedReq.abortResponse(HttpStatusException.of(HttpStatus.REQUEST_ENTITY_TOO_LARGE, cause),
-                                         true);
+            final ContentTooLargeException cause =
+                    ContentTooLargeException.builder()
+                                            .maxContentLength(maxContentLength)
+                                            .contentLength(req.headers())
+                                            .transferred(transferredLength)
+                                            .build();
+
+            final HttpStatusException httpStatusException =
+                    HttpStatusException.of(HttpStatus.REQUEST_ENTITY_TOO_LARGE, cause);
+            if (req.needsAggregation()) {
+                assert !req.isInitialized();
```

Comment:
Should we use `decodedReq` for consistency?
```suggestion
            if (decodedReq.needsAggregation()) {
                assert !decodedReq.isInitialized();
```